### ISSUE TITLE
CRAS cant be compiled

### ISSUE THREAD

**Issue by KuteKetX** [dnschneid/crouton#3811]
```
Compiling CRAS (i386)...
tests/cras_test_client.c: In function 'audio_debug_info':
tests/cras_test_client.c:585:10: error: format '%lu' expects argument of type 'long unsigned int', but argument 2 has type 'long long unsigned int' [-Werror=format=]
   printf("stream: %lu dev: %u\n",
          ^
tests/cras_test_client.c: In function 'run_aecdump':
tests/cras_test_client.c:677:10: error: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'uint64_t {aka long long unsigned int}' [-Werror=format=]
   printf("Dumping AEC info to %s, stream %lu, fd %d\n",
          ^
tests/cras_test_client.c: In function 'main':
tests/cras_test_client.c:1545:3: error: 'stream_id' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   run_aecdump(client, stream_id, 1);
   ^
tests/cras_test_client.c: At top level:
cc1: error: unrecognized command line option '-Wno-int-in-bool-context' [-Werror]
cc1: all warnings being treated as errors
/usr/lib/gcc/x86_64-linux-gnu/5/../../../i386-linux-gnu/crt1.o: In function `_start':
(.text+0x18): undefined reference to `main'
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...
chronos@localhost / $ god damnit...
```
whenever i try to install a chroot it always ends with this, ive deleted and reinstalled crouton (after assuming it was crouton) but it seems like more, anyone else ever had this issue?

and how did i cause this? i just used crouton to install xenial the normal way.

the command i used:
```
sudo sh -e ~/Downloads/crouton -t xorg,gtk-extra,cli-extra,core
```

**Comment by KuteKetX** [comment]
day 2, still having the same issue, anyone gonna help?

btw @mareedsmdphd are you still having the same issue?

**Comment by DennisLfromGA** [comment]
I just ran the following command and had no problems, the installation finished successfully:

- `sudo sh ~/Downloads/crouton -n CRAS -t xorg,gtk-extra,cli-extra,core`

Here are the two snippets I found pertaining to CRAS in the output:

> Compiling CRAS (i386)...
Installing CRAS...

> Compiling CRAS (native)...
Installing CRAS...

I ran this on my Acer C710 running FlintOS because that's what I was using at the time.
Here's the `croutonversion` output:

```
chronos@localhost ~/Downloads $ sudo edit-chroot -l CRAS
name: CRAS
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/CRAS...
crouton: version 1-20180611074759~master:3697b784
release: xenial
architecture: amd64
xmethod: xorg
targets: xorg,gtk-extra,cli-extra,core
host: version unknown
kernel: Linux localhost 4.4.64-09104-gffd9a4db8e11 #7 SMP Tue Dec 19 21:19:20 CST 2017 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/CRAS...
```

When I get a chance I'll try installing it on one or two of my other Chromebooks and followup here.

*Hope this helps,*
-**DennisLfromGA**

**Comment by mareedsmdphd** [comment]
Thanks, DennisLfromGA. Things here in Vegas were so very close.

chronos@localhost / $ sudo sh ~/Downloads/crouton -p '/media/removable/USB30a' -r buster -t kde_desktop,xiwi,chrome-dev -n buster3
Password: 
Invalid target "kde_desktop".
chronos@localhost / $ sudo sh ~/Downloads/crouton -p '/media/removable/USB30a' -r buster -t kde,xiwi,chrome-dev -n buster3
WARNING: USB booting is enabled; consider disabling it.
WARNING: Signed boot verification is disabled; consider enabling it.
You can use the following command: sudo crossystem dev_boot_usb=0 dev_boot_signed_only=1
Installing buster-amd64 chroot to /media/removable/USB30a/chroots/buster3
Downloading debootstrap 1.0.101...
################################################################################################################################################################# 100.0%################################################################################################################################################################# 100.0%
Patching debootstrap...
Downloading bootstrap files...
W: Cannot check Release signature; keyring file not available /usr/share/keyrings/debian-archive-keyring.gpg
I: Retrieving InRelease 
I: Retrieving Packages 
I: Validating Packages 
I: Resolving dependencies of required packages...
I: Resolving dependencies of base packages...
I: Checking component main on http://httpredir.debian.org/debian...
I: Retrieving libacl1 2.2.52-3+b1
I: Validating libacl1 2.2.52-3+b1
I: Retrieving adduser 3.117


and then:

he following packages were automatically installed and are no longer required:
  binutils binutils-common binutils-x86-64-linux-gnu gcc gcc-7 gcc-7-multilib gcc-multilib lib32asan4 lib32atomic1 lib32cilkrts5 lib32gcc-7-dev lib32gcc1 lib32gomp1
  lib32itm1 lib32mpx2 lib32quadmath0 lib32stdc++6 lib32ubsan0 libasan4 libasound2-dev:i386 libatomic1 libbinutils libc-dev-bin libc6-dev libc6-dev:i386 libc6-dev-i386
  libc6-dev-x32 libc6-i386 libc6-x32 libcc1-0 libcilkrts5 libdpkg-perl libgcc-7-dev libglib2.0-0 libitm1 liblsan0 libmpx2 libquadmath0 libspeex-dev:i386 libspeex1:i386
  libspeexdsp-dev:i386 libtsan0 libubsan0 libx32asan4 libx32atomic1 libx32cilkrts5 libx32gcc-7-dev libx32gcc1 libx32gomp1 libx32itm1 libx32quadmath0 libx32stdc++6
  libx32ubsan0 linux-libc-dev linux-libc-dev:i386 pkg-config
Use 'apt autoremove' to remove them.
Suggested packages:
  ed diffutils-doc
The following NEW packages will be installed:
  patch
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 123 kB of archives.
After this operation, 241 kB of additional disk space will be used.
Get:1 http://cdn-fastly.deb.debian.org/debian buster/main amd64 patch amd64 2.7.6-2 [123 kB]
Fetched 123 kB in 2s (65.8 kB/s)
Selecting previously unselected package patch.
(Reading database ... 15853 files and directories currently installed.)
Preparing to unpack .../patch_2.7.6-2_amd64.deb ...
Unpacking patch (2.7.6-2) ...
Setting up patch (2.7.6-2) ...
patch set to automatically installed.
Compiling CRAS (i386)...
tests/cras_test_client.c: In function 'audio_debug_info':
tests/cras_test_client.c:585:21: error: format '%lu' expects argument of type 'long unsigned int', but argument 2 has type 'long long unsigned int' [-Werror=format=]
   printf("stream: %lu dev: %u\n",
                   ~~^
                   %llu
          (unsigned long long)info->streams[i].stream_id,
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests/cras_test_client.c: In function 'run_aecdump':
tests/cras_test_client.c:677:44: error: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'uint64_t {aka long long unsigned int}' [-Werror=format=]
   printf("Dumping AEC info to %s, stream %lu, fd %d\n",
                                          ~~^
                                          %llu
tests/cras_test_client.c: In function 'main':
tests/cras_test_client.c:1545:3: error: 'stream_id' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   run_aecdump(client, stream_id, 1);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o: In function `_start':
(.text+0x28): undefined reference to `main'
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.
Unmounting /media/removable/USB30a/chroots/buster3...
chronos@localhost / $ 

But you've just given me an idea--I don't have to put 6 lb of meat in one giant ball.

Thanks, guys.

Mike.

**Comment by DennisLfromGA** [comment]
I just ran the following command on my Pixelbook and had no problems, the installation finished successfully:
- `sudo sh ~/Downloads/crouton -n cras -p /var/crouton -t xorg,gtk-extra,cli-extra,core`

Here are the snippets I found pertaining to CRAS in the output:
```
chronos@localhost ~/Downloads $ grep CRAS CRAS.script 
Fetching CRAS (branch 3a33a280aac17ceaac70220360dcc4688005dcee)...
Compiling CRAS (i386)...
Installing CRAS...
Compiling CRAS (native)...
Installing CRAS...
Audio from the chroot will now be forwarded to CRAS (Chromium OS audio server),
of CRAS. Should this happen, simply update your chroot.
```

This is the `screenfetch` output for reference:
```
chronos@localhost ~/Downloads $ screenfetch -E
             .,:loool:,.               chronos@localhost
         .,coooooooooooooc,.           OS: Chrome OS 
      .,lllllllllllllllllllll,.        Kernel: x86_64 Linux 4.4.138-14447-g7e90a0e25e5e
     ;ccccccccccccccccccccccccc;       Uptime: 3d 7h 42m
   'ccccccccccccccccccccccccccccc.     Packages: 0
  ,ooc::::::::okO0000OOkkkkkkkkkkk:    Shell: bash 4.3.48
 .ooool;;;;:xK0kxxxxxk0XK0000000000.   CPU: Intel Core i5-7Y57 @ 4x 3.3GHz [48.0°C]
 :oooool;,;OKdddddddddddKX000000000d   GPU: 
 lllllool;lNdllllllllllldNK000000000   RAM: 4257MiB / 7870MiB
 llllllllloMdcccccccccccoWK000000000  
 ;cllllllllXXc:::::::::c0X000000000d  
 .ccccllllllONkc;,,,;cxKK0000000000.  
  .cccccclllllxOOOOOOkxO0000000000;   
   .:cccccccclllllllloO0000000OOO,    
     ,:ccccccccclllcd0000OOOOOOl.     
       '::cccccccccdOOOOOOOkx:.       
         ..,::ccccxOOOkkko;.          
             ..,:dOkxl:.              
```

And here's the `croutonversion` output:
```
chronos@localhost ~/Downloads $ sudo edit-chroot -c /var/crouton/chroots -l cras
name: cras
encrypted: no
Entering /var/crouton/chroots/cras...
crouton: version 1-20180611074759~master:3697b784
release: xenial
architecture: amd64
xmethod: xorg
targets: xorg,gtk-extra,cli-extra,core
host: version 10820.0.0 (Official Build) dev-channel eve 
kernel: Linux localhost 4.4.138-14447-g7e90a0e25e5e #1 SMP PREEMPT Tue Jun 26 12:42:21 PDT 2018 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /var/crouton/chroots/cras...
```
-----
Since we can't get a `croutonversion` output could you please show us the first five lines displayed when you navigate to: [chrome://version](chrome://version)

With that have a few more details to see what we're dealing with.

*Hope this helps,*
-**DennisLfromGA**

**Comment by KuteKetX** [comment]
here @DennisLfromGA:
screenfetch:
```
chronos@localhost ~/Downloads $ sudo bash screenFetch-master/screenfetch-dev 
ls: cannot access '/var/db/pkg/*/*': No such file or directory
             .,:loool:,.               root@localhost
         .,coooooooooooooc,.           OS: Chrome OS 
      .,lllllllllllllllllllll,.        Kernel: x86_64 Linux 4.4.138-14463-ga3af6d2337eb
     ;ccccccccccccccccccccccccc;       Uptime: 1d 10h 12m
   'ccccccccccccccccccccccccccccc.     Packages: 0
  ,ooc::::::::okO0000OOkkkkkkkkkkk:    Shell: sudo
 .ooool;;;;:xK0kxxxxxk0XK0000000000.   CPU: Intel Celeron N2840 @ 2x 2.5823GHz
 :oooool;,;OKdddddddddddKX000000000d   GPU: 
 lllllool;lNdllllllllllldNK000000000   RAM: 3375MiB / 3865MiB
 llllllllloMdcccccccccccoWK000000000  
 ;cllllllllXXc:::::::::c0X000000000d  
 .ccccllllllONkc;,,,;cxKK0000000000.  
  .cccccclllllxOOOOOOkxO0000000000;   
   .:cccccccclllllllloO0000000OOO,    
     ,:ccccccccclllcd0000OOOOOOl.     
       '::cccccccccdOOOOOOOkx:.       
         ..,::ccccxOOOkkko;.          
             ..,:dOkxl:.              
chronos@localhost ~/Downloads $ 
```
croutonversion:
```
chronos@localhost ~/Downloads $ sudo sh -e ~/Downloads/crouton -V
crouton: version 1-20180611074759~master:3697b784
chronos@localhost ~/Downloads $ 
```

**Comment by DennisLfromGA** [comment]
@KuteKetX,

> Since we can't get a croutonversion output could you please show us the first five lines displayed when you navigate to: chrome://version

*Hope this helps,*
-**DennisLfromGA**

**Comment by mareedsmdphd** [comment]
So, I'm covering for a friend and had to drop by two different
hospitals--three patients and 6 residents. Headed home now, and plan to see
what's the issue.

(BTW, July 1 is when interns and residents start, so it's nice to avoid
teaching hospitals on this day.)

Anyhow, I'm not doing anything novel here; sure I've compiled CRAS without
issue many times. I'm on the Canary channel, so I don't find it remarkable.

My guess: the logic of timing and dependency enforcement may go south. New
compiler,? A subtle change might cause a cast to get skipped. I'm pretty
sure the folks at Google write more sophisticated code than I do. (Joe
Cello, Chris Date, Ted Codd, and I do know a bit about Relational Theory,
to hear one of us tell it.)

Suspect order of adding pkgs may be an issue. Real new stuff. I've only
managed a few non-trivial Sid installs.

I'll look in an hour. BTW, the screenfetch is just swell, I was going to
ask.

Mike


M A Reeds MD PhD DPhil (Oxon)
725-261-2802
mareeds@mareedsmdphd.net



On Jul 1, 2018 8:23 PM, "DennisL" <notifications@github.com> wrote:

@KuteKetX <https://github.com/KuteKetX>,

Since we can't get a croutonversion output could you please show us the
first five lines displayed when you navigate to: chrome://version

*Hope this helps,*
-*DennisLfromGA*

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/dnschneid/crouton/issues/3811#issuecomment-401661269>,
or mute the thread

**Comment by KuteKetX** [comment]
@DennisLfromGA i legit just put the output of crouton version at the bottom.

**Comment by DennisLfromGA** [comment]
@KuteKetX,

Nope, that's different.  
You gave us the version of the crouton _**[installer](https://github.com/dnschneid/crouton/blob/master/installer/crouton)**_ not the version of the chroot which is derived from the _**[croutonversion](https://github.com/dnschneid/crouton/blob/master/chroot-bin/croutonversion)**_ script in the chroot.  Since you can't successfully install a chroot I was trying to ascertain some of your Chromebook details by other means, namely: [chrome://version](chrome://version)

So, long story short: 
- _**please show us the first five lines displayed when you navigate to: chrome://version**_

-**DennisLfromGA**

**Comment by KuteKetX** [comment]
```

Google Chrome | 69.0.3476.0 (Official Build) canary (64-bit)
-- | --
Revision | 86d261c5d8d1e6cd27cd72f8887c8de670f11b48-refs/branch-heads/3476@{#1}
Platform | 10828.0.0 (Official Build) canary-channel gnawty
Firmware Version | Google_Gnawty.5216.239.156
Customization ID | ACER-OLAY
JavaScript | V8 6.9.237.1
Flash | 30.0.0.134 /opt/google/chrome/pepper/libpepflashplayer.so
User Agent | Mozilla/5.0 (X11; CrOS x86_64 10828.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3476.0 Safari/537.36
```

**Comment by DennisLfromGA** [comment]
@KuteKetX,

I suspect this may be a symptom of the 'canary' channel since @mareedsmdphd has reported it happening to him too, although I don't know if he has an Acer Chromebook 11 (C730 / C730E / C735) device also.  Maybe if some others could repo it we might be able to gather enough info. to determine the cause.  I couldn't repo it on my Pixelbook using the dev channel.

*Hope this helps,*
-**DennisLfromGA**

**Comment by CamdynS-CBC** [comment]
I filed this issue https://github.com/dnschneid/crouton/issues/3770 a while ago and it seems to be the same thing. No response, though. 

Eventually, the updates from Canary will reach Dev, Beta, and Stable channels, so not sure what the next course of action will be. 

My Model: (Dell Chromebook 11 CB1C13)

-CamdynS-

**Comment by KuteKetX** [comment]
it used to work fine and i dont remember my chromebook ever updating.

**Comment by DennisLfromGA** [comment]
So far, this only seems to be occurring on the 'canary' channel, if anyone is experiencing this issue on the beta or stable channel please let us know and show us the output of: `sudo edit-chroot -all`

*Hope this helps,*
-**DennisLfromGA**

**Comment by mareedsmdphd** [comment]
-DennisLfromGA, several  days ago, I reverted to the stable channel, and was able to update 4 chroot.

When I went to the dev channel, the issue reappeared. Going beta, no issues.

I've saved output, so I'll send it along in an hour or so.

Mike

**Comment by DennisLfromGA** [comment]
@mareedsmdphd,

I have two Chromebooks on the dev channel and have had no problems updating CrOS or my chroots in the past.  I haven't updated them to incorporate the latest merges but I'll do that hopefully tomorrow and report my results back here.

*Hope this helps,*
-**DennisLfromGA**

**Comment by mareedsmdphd** [comment]
Hi, @coderofsalvation , here's what I did so as not to lose things.

I copied the contents of my Downloads to a removable medium (USB 3.0 formatted to EXT 4)

I downloaded and ran the Chromebook Recovery Utility:

Check out "Chromebook Recovery Utility": https://chrome.google.com/webstore/detail/chromebook-recovery-utili/jndclpdbaamdhonoechobihbbiimdgai?utm_source=gmail
Oh, naturally, I did a sync. Little difficult because I was unable to do a wifi connect: most network interfaces were corrupted and irreparable. Used tethering with my Android.

Anyhow, I recovered to stable, updated and then went to beta.

No data loss, but not a trivial amount of time.

Let us know how that works.

Mike

**Comment by DennisLfromGA** [comment]
Yup, happenin' on the dev channel now too but the devs are on it. 👍

**Comment by mareedsmdphd** [comment]
@DennisLfromGA , @drinkcat , @dnschneid 👍 
Glad to see this closed. Was looking at Error Updating Chroot #3770, Error Updating Chroot #3770, and then https://bugs.chromium.org/p/chromium/issues/detail?id=864815, so I hope this went where I think it went.

[crhk201807171845.txt](https://github.com/dnschneid/crouton/files/2207940/crhk201807171845.txt)

Google Chrome | 68.0.3440.59 (Official Build) beta (64-bit)
-- | --
Revision | d2bc7dee8086bc46fe6153b4b8c45da429245503-refs/branch-heads/3440@{#644}
Platform | 10718.50.0 (Official Build) beta-channel banon
Firmware Version | Google_Banon.7287.373.0
ARC | 4883785
JavaScript | V8 6.8.275.21
Flash | 30.0.0.134 /opt/google/chrome/pepper/libpepflashplayer.so
User Agent | Mozilla/5.0 (X11; CrOS x86_64 10718.50.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.59 Safari/537.36
Command Line | /opt/google/chrome/chrome --ppapi-flash-path=/opt/google/chrome/pepper/libpepflashplayer.so --ppapi-flash-version=30.0.0.134 --ui-prioritize-in-gpu-process --use-gl=egl --enable-native-gpu-memory-buffers --gpu-sandbox-failures-fatal=yes --enable-logging --log-level=1 --use-cras --enable-wayland-server --user-data-dir=/home/chronos --system-developer-mode --login-profile=user --has-chromeos-keyboard --guest-wallpaper-large=/usr/share/chromeos-assets/wallpaper/guest_large.jpg --guest-wallpaper-small=/usr/share/chromeos-assets/wallpaper/guest_small.jpg --child-wallpaper-large=/usr/share/chromeos-assets/wallpaper/child_large.jpg --child-wallpaper-small=/usr/share/chromeos-assets/wallpaper/child_small.jpg --default-wallpaper-large=/usr/share/chromeos-assets/wallpaper/oem_large.jpg --default-wallpaper-small=/usr/share/chromeos-assets/wallpaper/oem_small.jpg --default-wallpaper-is-oem --enable-consumer-kiosk --arc-availability=officially-supported --enterprise-enrollment-initial-modulus=15 --enterprise-enrollment-modulus-limit=19 --login-manager --vmodule=*arc/*=1,*chromeos/login/*=1,auto_enrollment_controller=1,*/ui/ozone/*=1,*/ui/display/manager/chromeos/*=1,*night_light*=1,power_button_observer=2,webui_login_view=2,lock_state_controller=2,webui_screen_locker=2,screen_locker=2
Executable Path | /opt/google/chrome/chrome
Profile Path | /home/chronos/u-4a3c10609bf6f592dc377951c063b4cb460384eb
Variations | 6a89113b-ccb9a6a5d01ab0d3-f23d1dea59aeb88e-3f4a17df34a6bf44-ca7d8d8064da5c1e-f23d1deab1681d28-1410f109041608a-3f4a17df38eb801c-3f4a17df7c1bc906-f55a797447e5d3db-3d47f4f4125b7f68-39db89b5d442dfb7-eeca42f71149accc-f23d1dea4dc30737-b8a5ea08a582a1b8-ad75ce173042ad4b-ca54bb4798be3390-d93a0620116c6887-d7bc6f4544827ee5-f23d1deaedbcf7c5-51ca58f35485fc4d-ca7d8d809773d3bd-f23d1dea93731dca-cdcbcaf49e5c75f1-ee212b16f79cb77b-3f4a17df4ea303a6-f5d1e975bcc34a89-3f4a17df6e6e0c7e-70658ae4d92562a9-cfe3c2ea4da5ae82-3f4a17df2c1d398c-3f4a17df6973a1cf-3f4a17df4932440-d21eb72dff29b1bd-54dd8886da460ac8-3f4a17df4bc337ce-69465896d1466cda-3f4a17df9a2f4e5b-ca7d8d8017507c76-3d47f4f4494d8760-52325d433ac60855-486e2a9cf296190c-652559964442aae2-7158671eed1d377-e1cc0f1412e17bc5-e1cc0f1475f0f0a0-6bdfffe7e2b18481-4ad60575e7e71889-4ad60575469cffe2-21549ebe3a4029d-f23d1dea8834fcca-cf4f6ead81c6897f-3f4a17df

So, I'll prepare a recovery image, slog some data, and rejoin the dev channel.

Guys are great, and really smart.

As all of you knew when you woke up this morning.

@mareedsmdphd

**Comment by rudedog830** [comment]
Downloaded the code with latest commits.  Still experiencing this error:

Compiling CRAS (native)...
tests/cras_test_client.c: In function 'run_aecdump':
tests/cras_test_client.c:678:10: error: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'uint64_t' [-Werror=format=]
          aecdump_file, stream_id, aecdump_fd);
          ^
cc1: all warnings being treated as errors
/usr/lib/gcc/arm-linux-gnueabihf/4.8/../../../arm-linux-gnueabihf/crt1.o: In function `_start':
(.text+0x28): undefined reference to `main'
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.

Output of chrome://version:

Google Chrome | 69.0.3494.0 (Official Build) dev (32-bit)
-- | --
Revision | e91414c45bcdc6397f1a38faa0c826fbc47cd772-refs/branch-heads/3494@{#1}
Platform | 10888.0.0 (Official Build) dev-channel veyron_minnie
Firmware Version | Google_Veyron_Minnie.6588.237.0
ARC | 4898482

**Comment by CamdynS-CBC** [comment]
Agreed, still having a bit of an issue here. 

Google Chrome | 70.0.3502.0 (Official Build) canary (64-bit)
-- | --
Revision | bd32163382ec789e29eeceaf695587d76d77eb36-refs/branch-heads/3502@{#1}
Platform | 10918.0.0 (Official Build) canary-channel wolf
Firmware Version | Google_Wolf.4389.24.62

Compiling CRAS (i386)...
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc.a when searching for -lgcc
/usr/bin/ld: cannot find -lgcc
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc_s.so when searching for -lgcc_s
/usr/bin/ld: cannot find -lgcc_s
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.
Unmounting /media/removable/USB Drive/chroots/xenial...
Sending SIGTERM to processes under /media/removable/USB Drive/chroots/xenial...

-CamdynS-

**Comment by Gorvetco** [comment]
Yes, actually I have experienced this on all 4 channels, using both my Samsung Chromebook 4 and my Acer Chromebook 315. Most recently, Ive been experiencing this ongoing issue using my Acer. My preference is usually the dev channel. However, Ive now attempted installing crouton  via: 

`sudo crouton -r <xenial> -t x11,core,lxde,audio,keyboard,xiwi,cli-extra  -n xenial` 

while on all channels (canary, dev, beta, and now stable) All giving me the same output. It's rather frustrating honestly, I don't understand what the problem is. 
But, just to provide a bit more detail: 

I download crouton. like anyone normally would from https://goo.gl/fd3zc 

Then I Mount my sd card containing an ext4 fs to /media/SD
`mkdir /media/SD`
`sudo mount /dev/sdb /media/SD`

Then I create the "chroots" directories in /media/SD/chroots and I syslink it to /usr/local/chroots:
`mkdir /media/SD/chroots`
`sudo ln -s /media/SD/chroots /use/local/chroots`

Then I install crouton to /use/local/bin using:
`sudo install -Dt /usr/local/bin -m 755 ~/Downloads/crouton`

And then I attempt to `sudo crouton`  all day long... and I receive this as output ... all day long:

Compiling CRAS (i386)...
libcras/cras_client.c: In function 'request_floop_ready':
libcras/cras_client.c:1827:14: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   if (req == (struct floop_request *)tag)
              ^
libcras/cras_client.c: In function 'request_floop':
libcras/cras_client.c:4209:40: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  cras_fill_request_floop(&msg, params, (uint64_t)&req);
                                        ^
libcras/cras_client.c: At top level:
cc1: warning: unrecognized command line option '-Wno-int-in-bool-context'
/usr/bin/ld: cannot find -lcras
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.
The chroot setup script may be broken. Your chroot is not fully configured.
Removing the chroot setup script. You may want to update your chroot again.
UID 1000 not found in xenial
Unmounting /media/SD/chroots/xenial...

Any extra help would be much appreciated. I've also tried this with other Ubuntu versions (Jessie, trusty), and other desktop environments (such as e17, and xfce)

Thanks

**Comment by dnschneid** [comment]
See #4804
